src/core_plugins/kibana/public/context/query.js

```diff
@@ -0,0 +1,143 @@
+import _ from 'lodash';
+
+import { fetchAnchor } from './api/anchor';
+import { fetchPredecessors, fetchSuccessors } from './api/context';
+import { createSelector } from './utils/selectors';
+import { QueryParameterActionsProvider } from './query_parameters';
+
+
+function QueryActionsProvider($q, es, Private) {
```

Comment:
Then I would propose grouping them in a directory (`query/actions.js` and `query/selectors.js`?) though, because they are coupled via their knowledge of the state structure. If we had separate reducers I would keep reducers and selectors close, but now the reducer has been merged into the actions.